Modules/socketmodule.c

```diff
@@ -7628,6 +7633,14 @@ PyInit__socket(void)
     PyModule_AddIntMacro(m, AF_SYSTEM);
 #endif
 
+/* FreeBSD divert(4) */
+#ifdef PF_DIVERT
+    PyModule_AddIntMacro(m, PF_DIVERT);
+#endif
+#ifdef AF_DIVERT
+    PyModule_AddIntMacro(m, AF_DIVERT);
```

Comment:
It would be nice to add a comment why we need to check for both PF_DIVERT and
AF_DIVERT and mention the related freebsd versions.

You explained this in the discussion but having this in the code will make it much
easier to maintain 5 years from now.

Reply:
The distinction between **protocol families** (PF_* constants) and **address families** (AF_* constants) is what we inherited from the original BSD sockets API. The PF_FOO is supposed to be an argument for socket(2), while AF_FOO for any socket address manipulation syscalls or functions, e.g. bind(), getsockname(), etc. This distinction makes some sense, since existence of a protocol doesn't dictate existence of address type and vice versa. The divert(4) is actually nice example, as it uses AF_INET addresses. Or look at AF_RDS a couple lines above in the socketmodule.c - the same.

**However**, all modern implementations of sockets always declare both AF_FOO and PF_FOO. As far as I know, they always are equal. POSIX [doesn't mention PF_* families](https://pubs.opengroup.org/onlinepubs/9699919799/basedefs/sys_socket.h.html). Linux [notes their existence in the NOTES of the manual page](https://man7.org/linux/man-pages/man2/socket.2.html). Many software developers would use AF_FOO when calling socket(). This is why AF_DIVERT is also provided by FreeBSD and this is why I provided PyModule_AddIntMacro for it, too.

Reply:
The distinction between **protocol families** (PF_* constants) and **address families** (AF_* constants) is what we inherited from the original BSD sockets API. The PF_FOO is supposed to be an argument for socket(2), while AF_FOO for any socket address manipulation syscalls or functions, e.g. bind(), getsockname(), etc. This distinction makes some sense, since existence of a protocol doesn't dictate existence of address type and vice versa. The divert(4) is actually nice example, as it uses AF_INET addresses. Or look at AF_RDS a couple lines above in the socketmodule.c - the same.

**However**, all modern implementations of sockets always declare both AF_FOO and PF_FOO. As far as I know, they always are equal. POSIX [doesn't mention PF_* families](https://pubs.opengroup.org/onlinepubs/9699919799/basedefs/sys_socket.h.html). Linux [notes their existence in the NOTES of the manual page](https://man7.org/linux/man-pages/man2/socket.2.html). Many software developers would use AF_FOO when calling socket(). This is why AF_DIVERT is also provided by FreeBSD and this is why I provided PyModule_AddIntMacro for it, too.

Reply:
Right, looking at the rest of the code it is expected to have both constants.